services/sts/src/it/java/software/amazon/awssdk/services/sts/AssumeRoleIntegrationTest.java

```diff
@@ -221,4 +221,37 @@ public void profileCredentialsProviderCanAssumeRoles() throws InterruptedExcepti
         assertThat(awsCredentials.secretAccessKey()).isNotBlank();
         ((SdkAutoCloseable) awsCredentialsProvider).close();
     }
+
+    @Test
+    public void profileCredentialProviderCanAssumeRolesWithCredentialSource() throws InterruptedException {
```

Comment:
Hmm, this test failed on CodeBuild. Have you tested it locally?

```
ests run: 4, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 213.115 sec <<< FAILURE! - in software.amazon.awssdk.services.sts.AssumeRoleIntegrationTest 
profileCredentialProviderCanAssumeRolesWithCredentialSource(software.amazon.awssdk.services.sts.AssumeRoleIntegrationTest)  Time elapsed: 60.243 sec  <<< FAILURE! 
java.lang.AssertionError: Condition was not met after 18 attempts (60 seconds) 
    at software.amazon.awssdk.testutils.Waiter.orFailAfter(Waiter.java:163) 
    at software.amazon.awssdk.testutils.Waiter.orFail(Waiter.java:116) 
    at software.amazon.awssdk.services.sts.AssumeRoleIntegrationTest.profileCredentialProviderCanAssumeRolesWithCredentialSource(AssumeRoleIntegrationTest.java:248) 
```

Reply:
Yes, it succeeds on my machine. Does CodeBuild have `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables set when running tests? That could be an issue as the `credential_source = Environment` uses primarily the environment variables and system properties after that.

Reply:
The test uses system properties because those seemed easier to test against when I initially wrote the code. Now that I looked again there is `EnvironmentVariableHelper` which makes testing the environment variable case just as trivial. I can make the test explicitly unset the environment variables in the system properties test case if that's the issue.

Reply:
Yes, I think that's the case. On second thought, in our `DefaultCredentialsProvider` implementation, `SystemPropertyCredentialsProvider`  takes precedent over `EnvironmentVariableCredentialsProvider`.  Can we update `credential_source` to be consistent with that?